cmd/service/test.go

```diff
@@ -83,41 +84,53 @@ func executeTask(serviceID string, task string, dataPath string) (execution *cor
-	})
-	handleError(err)
+	var err error
+	serviceID := cmd.Flag("serviceID").Value.String()
+	if serviceID == "" {
+		service := loadService(defaultPath(args))
+		imageHash := buildDockerImage(defaultPath(args))
+		injectConfigurationInDependencies(service, imageHash)
+
+		deployment, err := cli.DeployService(context.Background(), &core.DeployServiceRequest{
+			Service: service,
+		})
+		handleError(err)
+		serviceID = deployment.ServiceID
+		fmt.Println("Service deployed with success with service ID:", serviceID)
 
-	cmdUtils.ShowSpinnerForFunc(cmdUtils.SpinnerOptions{Text: "Starting service..."}, func() {
-		_, err = cli.StartService(context.Background(), &core.StartServiceRequest{
-			ServiceID: deployment.ServiceID,
+		cmdUtils.ShowSpinnerForFunc(cmdUtils.SpinnerOptions{Text: "Starting service..."}, func() {
+			_, err = cli.StartService(context.Background(), &core.StartServiceRequest{
+				ServiceID: serviceID,
+			})
 		})
-	})
-	handleError(err)
-	fmt.Println(aurora.Green("Service started"))
+		handleError(err)
+		fmt.Println(aurora.Green("Service started"))
+	}
 
-	go listenEvents(deployment.ServiceID, cmd.Flag("event").Value.String())
-	go listenResults(deployment.ServiceID)
+	go listenEvents(serviceID, cmd.Flag("event-filter").Value.String())
+	go listenResults(serviceID, cmd.Flag("task-filter").Value.String(), cmd.Flag("output-filter").Value.String())
 
 	time.Sleep(time.Second)
-	executeTask(deployment.ServiceID, cmd.Flag("task").Value.String(), cmd.Flag("data").Value.String())
+	executeTask(serviceID, cmd.Flag("task").Value.String(), cmd.Flag("data").Value.String())
 	<-cmdUtils.WaitForCancel()
 
-	cmdUtils.ShowSpinnerForFunc(cmdUtils.SpinnerOptions{Text: "Stopping service..."}, func() {
-		_, err = cli.StopService(context.Background(), &core.StopServiceRequest{
-			ServiceID: deployment.ServiceID,
+	if cmd.Flag("keep-alive").Value.String() != "true" {
+		cmdUtils.ShowSpinnerForFunc(cmdUtils.SpinnerOptions{Text: "Stopping service..."}, func() {
+			_, err = cli.StopService(context.Background(), &core.StopServiceRequest{
+				ServiceID: serviceID,
+			})
 		})
-	})
-	handleError(err)
-	fmt.Println(aurora.Green("Service stopped"))
+		handleError(err)
+		fmt.Println(aurora.Green("Service stopped"))
+	}
 }
 
 func init() {
-	Test.Flags().StringP("event", "e", "*", "Only log a specific event")
-	Test.Flags().StringP("task", "t", "", "Run a specific task")
+	Test.Flags().StringP("event-filter", "e", "*", "Only log the data of the given event")
+	Test.Flags().StringP("task", "t", "", "Run the given task")
 	Test.Flags().StringP("data", "d", "", "Path to the file containing the data required to run the task")
+	Test.Flags().StringP("task-filter", "r", "", "Only log the result of the given task")
+	Test.Flags().StringP("output-filter", "o", "", "Only log the data of the given output of a task result. If set, you also need to set the task in --result")
+	Test.Flags().StringP("serviceID", "s", "", "ID of a previously deployed service")
```

Comment:
Flags `event`, `result`, `output` and `service` doesn't match with the API.
I renamed them with the API names.